
Code from 3rd class on python after orientation by LUMS course - Data Science 
and Machine Learning Using Python

### **IMPORTING MLB DATA FROM GITHUB**

We have data with 3 columns representing height (in inches), weight (in pounds) and age (in years). It’s stored in GitHub as a .CSV file.
We will import this dataset using the read function from the Panda’s Library; We will create a 2d Numpy array using this CSV File

In [9]:
#importing libraries
import pandas as pd
import numpy as np

#importing MLB data using the read function from Panda’s
baseball = pd.read_csv("https://raw.githubusercontent.com/Masadn/PythonCourse/master/dataset/mlbData.csv")

print(baseball)
print(type(baseball))

      Height  Weight    Age
0         74     180  22.99
1         74     215  34.69
2         72     210  30.78
3         72     210  35.43
4         73     188  35.71
...      ...     ...    ...
1010      73     190  25.08
1011      74     180  25.73
1012      75     205  25.19
1013      75     190  31.01
1014      73     195  27.92

[1015 rows x 3 columns]
<class 'pandas.core.frame.DataFrame'>


In [10]:
#creating a NumPy array
np_baseball = np.array(baseball)

In [11]:
print(np_baseball.shape)

(1015, 3)


**Task**

You want to convert the units of height and weight to metric (meters and kilograms respectively). As a first step, create a numpy array with three
values: (0.0254, 0.453592 and 1). Name this array np_conversion.
Multiply np_baseball with np_conversion and print out the result.

In [7]:
conversion=[0.0254, 0.453592, 1]
np_conversion=np.array(conversion)
print(type(np_conversion))

<class 'numpy.ndarray'>


In [12]:
np_result=np_baseball*np_conversion
print(np_result)

[[ 1.8796  81.64656 22.99   ]
 [ 1.8796  97.52228 34.69   ]
 [ 1.8288  95.25432 30.78   ]
 ...
 [ 1.905   92.98636 25.19   ]
 [ 1.905   86.18248 31.01   ]
 [ 1.8542  88.45044 27.92   ]]


**STATISTICS – AVERAGE VS MEDIAN**

In [13]:
x = [1, 4, 8, 10, 12]
x1 = np.array(x)
np.mean(x1)

7.0

In [15]:
np.median(x1)

8.0

In [16]:
np.sum(x1)

35

In [20]:
np.min(x1)

1

In [21]:
np.max(x1)

12

**MLB DATA SUMMARY STATISTICS - TASK BACKGROUND**

The baseball data is available as a 2D numpy array with 3 columns (height, weight,
age) and 1015 rows.
The name of this numpy array is np_baseball. After restructuring the data, however,
you notice that some height values are abnormally high.
Let’s try using some summary statistics on the MLB Dataset.

Print out the mean and median of players height from np_baseball_conversion


In [25]:
print(np_result[:,0])

[1.8796 1.8796 1.8288 ... 1.905  1.905  1.8542]


In [30]:
# print mean height
np_height=np_result[:,0]
print(np.mean(np_height))

1.8717172413793102


In [31]:
# print median height
print(np.median(np_height))

1.8796


**STATISTICS – STANDARD DEVIATION**

In [32]:
x = [1, 4, 8, 10, 12]
np.std(x)

4.0

**Task**
Use np.std() on the first column (height) of np_baseball_conversion to calculate standard deviation.
Print out standard deviation using print()

In [33]:
print(np.std(np_result[:,0]))

0.05874491377858227


**Task**

Use the code below to create heights and positions from FIFA dataset from github.
import pandas as pd
fifa = pd.read_csv("https://raw.githubusercontent.com/Masadn/PythonCourse/master/dataset/fifapos.csv")
◼ Convert heights and positions, which are regular lists, to numpy arrays.
◼ Call them np_heights and np_positions.
◼ Print out both np_heights and np_positions.

In [34]:
import pandas as pd
pd_fifa=pd.read_csv("https://raw.githubusercontent.com/Masadn/PythonCourse/master/dataset/fifapos.csv")
print(pd_fifa.head())

   position   height
0        GK      191
1         M      184
2         A      185
3         D      180
4         M      181


In [41]:
pd_fifa.columns

Index([' position', ' height'], dtype='object')

In [57]:
np_fifa=np.array(pd_fifa)
print(np_fifa)

[[' GK' 191]
 [' M' 184]
 [' A' 185]
 ...
 [' D' 183]
 [' D' 179]
 [' M' 179]]


In [60]:
np_fifa[:,1]

array([191, 184, 185, ..., 183, 179, 179], dtype=object)

In [62]:
np_heights=np_fifa[:,1]
print(np_heights)
print(type(np_heights))

[191 184 185 ... 183 179 179]
<class 'numpy.ndarray'>


In [63]:
np_fifa[:,0]

array([' GK', ' M', ' A', ..., ' D', ' D', ' M'], dtype=object)

In [65]:
np_positions=np_fifa[:,0]
print(np_positions)
print(np_positions.shape)

[' GK' ' M' ' A' ... ' D' ' D' ' M']
(8847,)


In [71]:
gk_height=np_fifa[np_positions==' GK']
print(gk_height)

[[' GK' 191]
 [' GK' 193]
 [' GK' 190]
 ...
 [' GK' 182]
 [' GK' 194]
 [' GK' 196]]


In [72]:
gk_height1=gk_height[:,1]
gk_height1

array([191, 193, 190, 185, 188, 187, 189, 196, 190, 189, 188, 189, 188,
       188, 190, 193, 181, 191, 193, 190, 183, 183, 193, 193, 196, 184,
       190, 186, 195, 191, 187, 185, 186, 185, 186, 190, 185, 185, 187,
       188, 188, 186, 187, 193, 183, 185, 190, 186, 182, 191, 194, 191,
       194, 188, 185, 189, 187, 188, 184, 176, 187, 186, 191, 188, 191,
       192, 190, 188, 189, 184, 180, 191, 197, 189, 185, 191, 190, 190,
       185, 186, 186, 189, 193, 182, 189, 188, 190, 191, 183, 189, 189,
       190, 188, 184, 194, 188, 194, 193, 186, 186, 188, 187, 191, 192,
       196, 187, 187, 187, 192, 184, 189, 184, 185, 187, 202, 190, 185,
       182, 188, 188, 195, 187, 183, 197, 188, 190, 193, 191, 187, 184,
       183, 185, 193, 180, 191, 185, 188, 188, 186, 188, 188, 184, 186,
       189, 193, 194, 183, 178, 184, 190, 183, 184, 188, 194, 190, 183,
       194, 188, 180, 173, 187, 183, 184, 193, 188, 183, 190, 191, 199,
       186, 198, 188, 189, 189, 185, 193, 191, 188, 185, 185, 18

In [73]:
gk_height1.shape

(930,)

**Task**

In [76]:
attack_heights=np_fifa[np_positions==' A']
print(attack_heights)

[[' A' 185]
 [' A' 186]
 [' A' 187]
 ...
 [' A' 175]
 [' A' 189]
 [' A' 175]]
